core/src/main/java/org/elasticsearch/search/controller/SearchPhaseController.java

```diff
@@ -410,6 +412,19 @@ public InternalSearchResponse merge(ScoreDoc[] sortedDocs, AtomicArray<? extends
             }
         }
 
+        //Collect profile results
+        InternalProfileShardResults shardResults = null;
+        if (!queryResults.isEmpty()) {
```

Comment:
one if would be enough `if (!queryResults.isEmpty() && firstResult.profileResults() != null)`